RoboCat/Src/Main.cpp

```diff
@@ -1,14 +1,396 @@
+			{
+				std::cout << std::endl;
+				std::cout << "\n[" << incomingAddress.ToString().c_str() << "] " << receivedMsg << std::endl;
+				std::cout << std::endl;
+				std::cout << "Message: ";
+			}
+		}
+	});
+
+	/*std::cout << "Press enter to exit at any time!\n";
+	std::cin.get();
+	quit = true;
+	connSocket->~TCPSocket(); // Forcibly close socket (shouldn't call destructors like this -- make a new function for it!*/
+	receiveThreadServer.join();
+	sendThreadServer.join();
+}
+
+void DoTcpClient(std::string port)
+{
+	// Create socket
+	TCPSocketPtr clientSocket = SocketUtil::CreateTCPSocket(SocketAddressFamily::INET);
+	if (clientSocket == nullptr)
+	{
+		SocketUtil::ReportError("Creating client socket");
+		ExitProcess(1);
+	}
+
+	LOG("%s", "Client socket created");
+
+	// Bind() - "Bind" socket -> tells OS we want to use a specific address
+
+	std::string address = StringUtils::Sprintf("127.0.0.1:%s", port.c_str());
+	SocketAddressPtr clientAddress = SocketAddressFactory::CreateIPv4FromString(address.c_str());
+	if (clientAddress == nullptr)
+	{
+		SocketUtil::ReportError("Creating client address");
+		ExitProcess(1);
+	}
+
+	if (clientSocket->Bind(*clientAddress) != NO_ERROR)
+	{
+		SocketUtil::ReportError("Binding client socket");
+		// This doesn't block!
+		ExitProcess(1);
+	}
+
+	LOG("%s", "Bound client socket");
+
+	// Connect() -> Connect socket to remote host
+
+	SocketAddressPtr servAddress = SocketAddressFactory::CreateIPv4FromString("127.0.0.1:8080");
+	if (servAddress == nullptr)
+	{
+		SocketUtil::ReportError("Creating server address");
+		ExitProcess(1);
+	}
+
+	if (clientSocket->Connect(*servAddress) != NO_ERROR)
+	{
+		SocketUtil::ReportError("Connecting to server");
+		ExitProcess(1);
+	}
+
+	LOG("%s", "Connected to server!");
+
+	// Enter chat text
+	std::cout << "\n\n\n--------------- CHAT ROOM --------------- \n\n\n" << std::endl;
+
+	// Get username
+	std::string username;
+
+	std::cout << "Enter username: ";
+	std::getline(std::cin, username);
+
+	std::cout << "USERNAME SET TO: " << username << std::endl;
+
+	std::cout << "\nCommands: \n /exit - quits program \n Pressing Enter Sends Message\n";
+	std::cout << std::endl;
 
+	bool quit = false;
+	bool sendQuit = false;
+
+	std::thread sendThreadClient([&]() { // don't use [&] :)
+		std::cout << "Message: ";
+
+		while (!sendQuit) // Need to add a quit here to have it really exit!
+		{
+			std::string msg;
+			std::getline(std::cin, msg);
+
+			if (msg == "/exit")
+			{
+				sendQuit = true;
+				//clientSocket->~TCPSocket();
+				clientSocket->Send(msg.c_str(), msg.length());
+				std::cout << "User has disconnected.";
+				ExitProcess(1);
+				break;
+			}
+			else
+			{
+				msg = username + ": " + msg;
+				clientSocket->Send(msg.c_str(), msg.length());
+				std::cout << std::endl;
+				std::cout << "Message: ";
+			}
+		}
+		});
+
+	std::thread receiveThreadClient([&]() { // don't use [&] :)
+		while (!quit) // Need to add a quit here to have it really exit!
```

Comment:
Might as well have made this `while (true)`, since `quit` is never changed and the loop is broken through `return` and `ExitProcess()`